### ISSUE TITLE
"mode" property for Responsive breakpoints

### ISSUE THREAD

**Issue by htopia** [tsparticles/tsparticles#2358]
### Contact Details

htopia.game@gmail.com

### What is your request?

The current responsive option only accepts breakpoints that validate the maxWidth property against the canvas size. This works fine when using the fullscreen mode because the canvas size is the same as the screen size.

However, when using tsparticles in just specific containers of an application, the containers may take different dimensions based on CSS media queries, making the tsparticles' canvas size too dynamic and difficult to control with the maxWidth property of the responsive breakpoints.

I propose a property "mode" for each responsive breakpoint config, so that we can specify if we want to validate the breakpoint against the canvas size or against the screen size.

Example:

```
{
    "particles": {
        ...otherProperties,
        "size": {
            "value": 5
        }
    },
    "responsive": [
        {
            "maxWidth": 1200,
            "mode": "screen", //"canvas" | "screen" (it could default to "canvas")
            "options": {
                "particles": {
                    "size": {
                        "value": 2
                    }
                }
            }
        }
    ]
}
```

This way anyone can specify how they want their breakpoints validated against.

Hope this makes sense.

Thanks!

### Version

1.36

### Which library are you using?

Vanilla (tsparticles), React.js (react-tsparticles, react-particles-js)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Guzzler** [comment]
I would like to pick this up @matteobruni should i submit a PR for the same?

**Comment by matteobruni** [comment]
> I would like to pick this up @matteobruni should i submit a PR for the same?

Yes you have to submit a PR

**Comment by matteobruni** [comment]
Fixed in `v1.37.0`, thanks to @Guzzler
